spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/TypeConverter.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2023 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.spark.bigquery;
+
+import com.google.cloud.bigquery.LegacySQLTypeName;
+import com.google.protobuf.DescriptorProtos;
+import org.apache.spark.sql.types.DataType;
+
+public interface TypeConverter<T> {
```

Comment:
Do we need this to be public?

Reply:
I think the idea is to let users have the option to bring their own type converters. cc @davidrabinowitz

Reply:
Yes, this needs to be public for implementing different type of converters in different versions